### PR TITLE
feat: Add custom loss objective support.

### PR DESCRIPTION
**Pull request by OctopusTakopi**

### CODE REVIEW

File: examples/custom_loss_regression.rs [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+        {
+            "num_iterations": 100,
+            "early_stopping_rounds": 10,
+            "objective": "regression",
+            "metric": "l2",
+            "learning_rate": 0.05,
+            "num_leaves": 31,
+            "verbose": -1
+        }
+    };
```
Comment (Mottl) [inline examples/custom_loss_regression.rs]: Do we actually need both `params` and `builtin_params`? (same thing in `custom_loss_multiclass.rs`)

**Review by Mottl — COMMENTED** [review]
Great job! Thanks a lot!

**Comment by OctopusTakopi** [comment]
should be good now

**Comment by Mottl** [comment]
Great! Thanks a lot one more time
